### ISSUE TITLE
https upstream server requires client authentication

### ISSUE THREAD

**Issue by andihess** [fabiolb/fabio#459]
Does fabio support a scenario where an https upstream server requires client authentication?
The certificate to be used by fabio for the client authentication is provided by the client that hits fabio already.

**Comment by magiconair** [comment]
No, but I don't think this should be difficult to add. According to https://golang.org/pkg/crypto/tls/#Config the TLS client needs to either set `Certificates` or `GetClientCertificate`. 

The `http.Transport`s would need a `GetClientCertificate` function which would need to be configured here:

https://github.com/fabiolb/fabio/blob/master/main.go#L171-L186

This should probably be provided by a cert store and could be similar to this:

https://github.com/fabiolb/fabio/blob/master/cert/source.go#L89-L151

If you want to submit a PR for this I'd be more than happy to merge it.
